### PR TITLE
Add `CudaState` as a wrapper around cudarc primitives

### PR DESCRIPTION
**Pull request by MegaRedHand**

### CODE REVIEW

File: gpu/src/cuda/abstractions/state.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+        // Load PTX libraries
+        // TODO: extract into helper method
+        device
+            .load_ptx(
+                Ptx::from_src(/* Library code */ FFT_PTX),
+                /* Library name */ "fft",
+                /* defined functions */ &["radix2_dit_butterfly"],
+            )
+            .map_err(|err| CudaError::PtxError(err.to_string()))?;
```
Comment (matias-gonz) [inline gpu/src/cuda/abstractions/state.rs:31]: Will we load all modules (`fft`, `gen_twiddles` and `bitrev_permutation`) at new()?

File: gpu/src/cuda/fft/polynomial.rs [hunk 2]
```diff
@@ -17,10 +17,11 @@ where
     F: IsFFTField,
     F::BaseType: Unpin,
 {
+    let state = CudaState::new().unwrap();
```
Comment (matias-gonz) [inline gpu/src/cuda/fft/polynomial.rs:20]: Should we remove this unwrap()?

File: gpu/src/cuda/abstractions/state.rs (hunk 1 above)
Comment (MegaRedHand) [reply gpu/src/cuda/abstractions/state.rs:29]: I was thinking of moving the module loading to the corresponding `get_<function>` implementations, but I think we should benchmark it first.

File: gpu/src/cuda/fft/polynomial.rs (hunk 2 above)
Comment (MegaRedHand) [reply gpu/src/cuda/fft/polynomial.rs]: Removed! Thanks for finding it!

File: gpu/src/cuda/abstractions/state.rs (hunk 1 above)
Comment (xqft) [reply gpu/src/cuda/abstractions/state.rs]: For now don't think too much about these kinds of optimizations, I think PTX loading is pretty small compared to what FFT takes to run

**Pull request merged**